src/application/embeddings/errors.py

```diff
@@ -0,0 +1,17 @@
+
+from src.constants import SUPPORTED_FILES_TUPLE
+
+
+class InvalidFileExtensionError(Exception):
+    """
+        Exception raised when a file does not have the expected extension.
+
+        Consider that only the following extensions are supported:
+        - archive files (.zip files)
```

Comment:
can we easily support .tar archive as well?

Reply:
Yup, gonna try that before to complete this pull request.

Reply:
I'm writing this here. 

- updated code to handle support based on content-type received inside the `FileParser` class
- now, not only `zip` files are supported, but also `tar` and `gz` (libraries for both archive files are included in Python, so that was easy)